flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultDeploymentValidator.java

```diff
@@ -43,8 +43,8 @@
 import java.util.Optional;
 import java.util.Set;
 
-/** Default validator implementation. */
-public class DefaultDeploymentValidator implements FlinkDeploymentValidator {
+/** Default validator implementation for {@link FlinkDeployment}. */
+public class DefaultDeploymentValidator implements FlinkResourceValidator<FlinkDeployment> {
```

Comment:
As we discussed in the other PR could we change this to explicitly have 2 validation methods, one for SessionJob and one for Deployment ? And remove the generic logic

Reply:
Yes, I will do it.